cli/commands/site/create.ts

```diff
@@ -289,6 +277,31 @@ export const registerCommand = ( yargs: StudioArgv ) => {
 					type: 'string',
 					describe: __( 'WordPress version (e.g., "latest", "6.4", "6.4.1")' ),
 					default: DEFAULT_VERSIONS.wp,
+					coerce: ( value: string ) => {
+						if ( ! isValidWordPressVersion( value ) ) {
+							throw new LoggerError(
+								generateYargsErrorMessage(
+									'wp',
+									value,
+									__(
+										'"latest", "nightly", or a valid version number (e.g., "6.4", "6.4.1", "6.4-beta1")'
+									)
+								)
+							);
```

Comment:
```suggestion
							throw new ValidationError(
								'wp',
								value,
								__(
									'Must be: "latest", "nightly", or a valid version number (e.g., "6.4", "6.4.1", "6.4-beta1")'
								)
							);
```
A couple of thoughts:
1. Let's use a new Error class in place of `generateYargsErrorMessage`.
2. Let's stick with complete, translatable sentences in `__()` calls

Reply:
Great idea with class and agree about complete messages 👍